src/ui/src/app/repositories/repositories.component.ts

```diff
@@ -0,0 +1,101 @@
+import { Component, OnInit } from '@angular/core';
+import { ReposService } from '../shared/services/repos.service';
+import { Repo } from '../shared/models/repo';
+import { Router } from '@angular/router';
+import { ConfigService } from '../shared/services/config.service';
+import { DomSanitizer } from '@angular/platform-browser';
+import { MdDialogRef, MdDialog, MdDialogConfig, MdIconRegistry, MdSnackBar } from '@angular/material';
+import { DialogsService } from '../shared/services/dialogs.service';
+import { RepositoryNewComponent } from '../repository-new/repository-new.component';
+
+@Component({
+  selector: 'app-repositories',
+  templateUrl: './repositories.component.html',
+  styleUrls: ['./repositories.component.scss'],
+  viewProviders: [MdIconRegistry]
+})
+export class RepositoriesComponent implements OnInit {
+  loading: boolean = true;
+  repos: Repo[] = [];
+  
+  constructor(
+    private reposService: ReposService,
+    private router: Router,
+    private config: ConfigService,
+    private mdIconRegistry: MdIconRegistry,
+    private sanitizer: DomSanitizer,
+    private dialogsService: DialogsService,
+    private dialog: MdDialog,
+    public snackBar: MdSnackBar,
+  ) {}
+
+  ngOnInit() {
+    this.mdIconRegistry.addSvgIcon(
```

Comment:
Are we using the `search`, `close` and `menu` icons? I only see the `delete` icon in the template